sds_data_manager/lambda_code/SDSCode/batch_starter.py

```diff
@@ -437,24 +505,64 @@ def lambda_handler(event: dict, context):
 
         # Start Batch Job execution for those that has all dependencies
         for downstream_data in downstream_instruments_to_process:
-            command = downstream_data["command"]
-            logger.info(f"Submitting job with this command - {command}")
+            # FYI, these are the keys the upstream_dependencies should contain:
+            # {
+            #     'instrument': 'swe',
+            #     'data_level': 'l0',
+            #     'descriptor': 'lveng-hk',
+            #     'start_date': '20231212',
+            #     'version': 'v001',
+            # },
+            batch_command = [
+                "--instrument",
+                downstream_data["instrument"],
+                "--data-level",
+                downstream_data["data_level"],
+                "--descriptor",
```

Comment:
Does "descriptor" now always include info from query_upstream_dependencies, meaning that the dependencies table will provide descriptor for all levels and instruments because it's a required field there? Does that mean the descriptor will be "all" if that's the descriptor in the dependencies table?

Reply:
yes. 'all' descriptor would be used to kick of certain level which will produce multiple files from within processing code.

We had to add 'descriptor' because without that, status tracking table will only have instrument, level, startdate, version and etc. which was causing row to be not unique initially and during lookup also.

Reply:
Yeah, I understand the issue. That sounds good to me!